### PR TITLE
thrift: exclude more library building

### PR DESCRIPTION
**Pull request by jimexist**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Review by danielnachun — CHANGES_REQUESTED** [review]
We need to add `uses_from_macos "zlib"` after the last `depends_on` line to fix the audit failure in Linux.  I'm not sure why this didn't fail before, as I see we've already bottled this.

**Comment by Jimexist** [comment]
> We need to add `uses_from_macos "zlib"` after the last `depends_on` line to fix the audit failure in Linux. I'm not sure why this didn't fail before, as I see we've already bottled this.

i don't think locally this command would fail:

```
❯ time brew audit --strict thrift
brew audit --strict thrift  3.30s user 1.25s system 94% cpu 4.825 total
```

maybe the compiling process no longer requires zlib?

**Comment by Jimexist** [comment]
either way i've added back the line @danielnachun and it seems that the warning is gone

**Comment by danielnachun** [comment]
> either way i've added back the line @danielnachun and it seems that the warning is gone

Yeah I just double checked the old bottle and it was already linked to `zlib` before, so it's not the case that we're adding or removing a link to anything.  

In that case we shouldn't need a revision bump - all we're doing is making sure that `thrift` doesn't opportunistically link to any of the many libraries it looks for while building.  Thanks for your contribution!

**Review by danielnachun — APPROVED** [review]

File: Formula/thrift.rb [hunk 1]
```diff
@@ -46,15 +47,24 @@ def install
       --prefix=#{prefix}
       --libdir=#{lib}
       --with-openssl=#{Formula["openssl@1.1"].opt_prefix}
-      --without-erlang
-      --without-haskell
```
Comment (carlocab) [inline Formula/thrift.rb:50]: Any reason why we're removing this?

Comment (danielnachun) [reply Formula/thrift.rb:50]: Whoops, didn't see those were being removed!  Sorry I merged already.  I can add back as a followup if needed, as I don't know why we'd remove these either.

Comment (danielnachun) [reply Formula/thrift.rb:50]: Actually, `--without-erlang` was just moved down, but `--without-haskell` seems gone now.

Comment (carlocab) [reply Formula/thrift.rb:50]: Yes, I was referring out `--without-haskell`?

Comment (Jimexist) [reply Formula/thrift.rb:50]: @danielnachun and @carlocab `haskell` was removed from thrift, same as common lisp:

- https://github.com/apache/thrift/tree/master/lib
- https://github.com/apache/thrift/tree/v0.16.0/lib
- https://issues.apache.org/jira/browse/THRIFT-5347

Comment (Jimexist) [reply Formula/thrift.rb:50]: for future references, the list (and its ordering) was copied from https://github.com/apache/thrift/blob/b664cfe2533e4bbf00fd5e7e0211bf7161ee2a04/configure.ac#L115-L133

Comment (danielnachun) [reply Formula/thrift.rb:50]: I see, so `--without-haskell` was just being ignored.  I think everything is okay then.

Comment (Jimexist) [reply Formula/thrift.rb:50]: affirmative, `--without-haskell` would have been ignored (unused flag) for 0.16.0 and `head` building.

**Pull request merged**